src/v/utils/mutex.h

```diff
@@ -28,25 +28,18 @@
  *    return m.with([] { ... });
  *    ```
  *
- * The `named_mutex` variant is analgous to `seastar::named_semaphore`. See
- * `seastar/core/semaphore.hh` for more information using an exception factory.
- *
  */
-template<
-  typename ExceptionFactory,
-  typename Clock = typename ss::timer<>::clock>
-class basic_mutex {
+class mutex {
 public:
-    using underlying_t = ss::basic_semaphore<ExceptionFactory, Clock>;
+    using underlying_t = ss::basic_semaphore<
+      ss::semaphore_default_exception_factory,
+      ss::timer<>::clock>;
```

Comment:
you can replace all `underlying_t` with `ss::semaphore` since we're using those default parameters anyway. or use `using underlying_t = ss::semaphore`.